src/5.0_to_5.1/calculated-measures/migrateCalculatedMeasures.ts

```diff
@@ -88,42 +101,84 @@ export function migrateCalculatedMeasures(
-                readers: ["ROLE_USER"],
-                timestamp: 1669281586947,
-                lastEditor: "admin",
-                canRead: true,
-                canWrite: true,
+      const cubeNames = measureToCubeMapping[measureName];
+
+      try {
+        // If there are no `cubeNames`, the calculated measure is not used in any saved widgets or dashboards.
+        if (!cubeNames) {
+          // The calculated measure was not migrated.
+          counters.calculated_measures.failed++;
+          _addErrorToReport(errorReport, {
+            contentType: "calculated_measures",
+            folderId,
+            folderName,
+            fileErrorReport: {
+              error: {
+                message: `Warning: Calculated measure "${measureName}" was not migrated because it is not currently used in any saved widgets or dashboards.`,
               },
-              children: {},
-            };
-          }
-          // `cmFolder.children[cubeName].children` is created above if it does not already exist.
-          cmFolder.children[cubeName].children![`[Measures].[${measureName}]`] =
-            record;
+            },
+            fileId: id,
+            name: measureName,
+          });
+          return;
         }
-      });
+
+        const migratedContent = migrateCalculatedMeasureContent(
+          JSON.parse(record.entry.content),
+          measureName,
+        );
+        record.entry.content = JSON.stringify(migratedContent);
+
+        cubeNames.forEach((cubeName) => {
+          if (cmFolder && cmFolder.children) {
+            // If `cubeName` property does not already exist, create it.
+            if (!cmFolder.children[cubeName]) {
+              cmFolder.children[cubeName] = {
+                entry: {
+                  isDirectory: true,
+                  owners: ["ROLE_USER"],
+                  readers: ["ROLE_USER"],
+                  timestamp: 1669281586947,
+                  lastEditor: "admin",
+                  canRead: true,
+                  canWrite: true,
+                },
+                children: {},
+              };
+            }
+            // `cmFolder.children[cubeName].children` is created above if it does not already exist.
+            cmFolder.children[cubeName].children![
+              `[Measures].[${measureName}]`
+            ] = record;
+          }
+        });
+        // The calculated measure was successfully migrated.
+        counters.calculated_measures.success++;
+      } catch (error) {
+        // The calculated measure was not migrated.
+        counters.calculated_measures.failed++;
+        _addErrorToReport(errorReport, {
+          contentType: "calculated_measures",
+          folderId,
+          folderName,
+          fileErrorReport: {
+            error: {
+              message: `Warning: Calculated measure "${measureName}" was not migrated because it is not currently used in any saved widgets or dashboards.`,
+            },
```

Comment:
This warning message is not appropriate anymore in the case of a generic error.
Rather, we must take the error itself.
```suggestion
            error: _serializeError(error, {
              doesReportIncludeStacks,
            }),
```